sim/firesim-lib/src/main/scala/bridges/TracerVBridge.scala

```diff
@@ -208,15 +208,62 @@ class TracerVBridgeModule(key: TracerVKey)(implicit p: Parameters)
     hPort.hBits.triggerDebit := !trigger && triggerReg
     hPort.hBits.triggerCredit := trigger && !triggerReg
 
-    val uint_traces = (traces map (trace => Cat(trace.valid, trace.iaddr).pad(64))).reverse
-    streamEnq.bits := Cat(uint_traces :+ trace_cycle_counter.pad(64)).pad(BridgeStreamConstants.streamWidthBits)
-
-    hPort.toHost.hReady := tFireHelper.fire(hPort.toHost.hValid)
     hPort.fromHost.hValid := tFireHelper.fire(hPort.fromHost.hReady)
 
-    streamEnq.valid := tFireHelper.fire(streamEnq.ready, trigger) && traceEnable
+    // the maximum widht of a single arm, this is determined by the 512 bit width of a single beat
+    val armWidth = 7
+    
+    // divide with a ceiling round, to get the total number of arms
+    val armCount = (traces.length + armWidth - 1) / armWidth
+    val armCountU = (armCount).U
```

Comment:
Parenthesis not necessary. I'd also prefer to remove this variable.

Reply:
Both armCount and armCountU are required for operation. Perhaps you have a better name I can use?

Reply:
generating the `UInt` literal in situ with `.U` is more idiomatic Chisel, so you should just do that.